tests/yz_crdt.erl

```diff
@@ -113,6 +141,387 @@ confirm() ->
+           {<<"update">>, register},
+           fun(R) ->
+                   riakc_register:set(<<"foo">>, R)
+           end, M1),
+    ok = riakc_pb_socket:update_type(
+           Pid,
+           Bucket,
+           Key,
+           riakc_map:to_op(M2)),
+    M3 = riakc_map:update(
+           {<<"update">>, register},
+           fun(R) ->
+                   riakc_register:set(<<"bar">>, R)
+           end, M1),
+    ok = riakc_pb_socket:update_type(
+           Pid,
+           Bucket,
+           Key,
+           riakc_map:to_op(M3)),
+
+    yokozuna_rt:commit(Cluster, Index),
+
+    {ok, {search_results, Results, _, _}} = riakc_pb_socket:search(
+                                              Pid, Index,
+                                              <<"update_register:*">>),
+
+    lager:info("Search update_register:*: ~p~n", [Results]),
+    ?assertEqual(1, length(Results)).
+
+%% @doc Tests varying deletes of within a CRDT map and checks for correct counts
+%%      - Remove registers, remove and add elements within a set
+%%      - Delete the map (associated w/ a key)
+%%      - Recreate objects in the map and delete the map again
+test_delete(Pid, Cluster, Bucket, Index, Key) ->
+    {ok, M1} = riakc_pb_socket:fetch_type(Pid, Bucket, Key),
+
+    lager:info("Remove register from map"),
+    M2 = riakc_map:erase({<<"name">>, register}, M1),
+
+    lager:info("Delete element from set (in map) & Add element to set (in map)"),
+    M3 = riakc_map:update(
+           {<<"interests">>, set},
+           fun(S) ->
+                   riakc_set:del_element(<<"thing">>,
+                   riakc_set:add_element(<<"roses">>, S))
+           end, M2),
+
+    ok = riakc_pb_socket:update_type(
+           Pid,
+           Bucket,
+           Key,
+           riakc_map:to_op(M3)),
+
+    yokozuna_rt:commit(Cluster, Index),
+
+    {ok, {search_results, Results1, _, _}} = riakc_pb_socket:search(
+                                              Pid, Index,
+                                              <<"name_register:*">>),
+    lager:info("Search deleted/erased name_register:*: ~p~n", [Results1]),
+    ?assertEqual(0, length(Results1)),
+
+    lager:info("Add another element to set (in map)"),
+    M4 = riakc_map:update(
+           {<<"interests">>, set},
+           fun(S) ->
+               riakc_set:add_element(<<"pans">>, S)
+           end, M3),
+
+    ok = riakc_pb_socket:update_type(
+           Pid,
+           Bucket,
+           Key,
+           riakc_map:to_op(M4)),
+
+    yokozuna_rt:commit(Cluster, Index),
+
+    {ok, {search_results, Results2, _, _}} = riakc_pb_socket:search(
+                                               Pid, Index,
+                                               <<"interests_set:thing*">>),
+    lager:info("Search deleted interests_set:thing*: ~p~n", [Results2]),
+    ?assertEqual(0, length(Results2)),
+
+    lager:info("Delete key for map"),
+    ?assertEqual(ok, riakc_pb_socket:delete(Pid, Bucket, Key)),
```

Comment:
Not sure it is fully in scope for this test (as this looks very yz-focused), but ideally a delete shall pass the context/vclock.

And it looks to me that at the moment it is not possible to delete a CRDT key with riakc passing the context/vclock, as the context/vclock seems not possible to be extracted (cleanly) from the value returned from riakc_pb_socket:fetch_type.